app/code/core/Mage/Catalog/Model/Resource/Product/Action.php

```diff
@@ -73,6 +73,8 @@ public function updateAttributes($entityIds, $attrData, $storeId)
                 }
                 $this->_processAttributeValues();
             }
+
+            $this->_updateUpdatedAt($entityIds);
```

Comment:
Should this happen after the `commit`?

Reply:
why? I think it should be committed together with the data, otherwise they could be out of sync

Reply:
Because the commit might have more observers - and might entirely cut the process alltogether. Why update update_at before we're really done?

Reply:
if the transaction fails, the commit fails and the updated_at doesn't get updated

Reply:
And the first part?
`Because the commit might have more observers`

Reply:
if the commit has an obsever that cuts the process, and the updatedAt is in the same transaction, then it work correctly (if i'm not missing anything). if the updatedAt is outside of the commit it will be executed also if the transaction fails, which is wrong.

Reply:
Forget about `cut the process`. `anything` is done in the `commit`. Maybe alter the data, maybe add more data, whatever. Shouldnt the `updated_at` be done `after` that is done, instead of before?
I am just proposing moving the lines, not stepping out of the try/catch.

As I read that, the code in `$this->_updateUpdatedAt` doesnt care about the `commit` or am I wrong?

Reply:
I see, but then we should check if something was committed or otherwise not update the updatedAt.

btw it seems quite wrong for the commit observer to add/remove more data than the one in process of being commited 😅

anyway let's see what other people say :-)

Reply:
Moving the lines shouldnt hurt and because the commit went through successfuly, we know for sure that we can update the updated_at field.
As per `but then we should check if something was committed or otherwise not update the updatedAt.` is that possible without knowing previous data vs. data after the commit? That sounds quite heavy.

Reply:
1. if the commit observer removes some data -> the updatedAt would be wrong
2. if the commit observer adds some data -> the updatedAt would be wrong
3. if the commit observer removed everything -> we would still update all the updatedAt

if it is so important to move the updatedAt outside of the commit, I prefer not to have it all like it is now in production.

Reply:
Unless there is **actual test case** that shows the update should be after the commit, I have to agree with @fballiano